### PR TITLE
CMR-4510: Modified api-doc to clarify the bulk updates on nested fiel…

### PR DESCRIPTION
**Pull request by sxu123**
…ds Platforms, Instruments and DataCenters.

### CODE REVIEW

**Comment by sxu123** [comment]
1. I have added the demos for all the update types for PLATFORMS, INSTRUMENTS and DATA_CENTERS bulk updates in sprint82.clj and merged it.
2. Found out that FIND_AND_REMOVE doesn't work for INSTRUMENTS bulk update.
3. The only wiki page I'm aware of for the bulk update is the design wiki page: https://wiki.earthdata.nasa.gov/display/CMR/CMR+Bulk+Update
Not sure if it's the right place to add a lot of examples for various update types.

File: ingest-app/docs/api.md [hunk 1]
```diff
@@ -581,40 +579,43 @@ The following update types are supported:
...
+  * Update field (required) - choose from the enumeration: SCIENCE_KEYWORDS, LOCATION_KEYWORDS, DATA_CENTERS, PLATFORMS, INSTRUMENTS
+  * Update value (required for all update types except for FIND_AND_REMOVE) - UMM-JSON representation of the update to make
+  * Find value (required for FIND_AND_REPLACE, FIND_AND_UPDATE and FIND_AND_REMOVE update types) - UMM-JSON representation of the data to find
+
+Update types that include a FIND will match on the fields supplied in the find value. For example, for a science keyword update with a find value of {"Category": "EARTH SCIENCES"}, any science keyword with a category of "EARTH SCIENCES" will be considered a match regardless of the values of the science keyword topic, term, etc.  It's worth noting that find value can not contain nested fields. So for bulk update on PLATFORMS, for example, find value can only contain Type, ShortName and LongName, not the nested fields like Characteristics and Instruments. On the other hand, update value can contain all the valid fields including the nested fields. So, nested fields can be updated, they just can't be used to find the matches. update value can not contain arrays.
```
Comment (chris-durbin) [inline ingest-app/docs/api.md]: EARTH SCIENCE singular.

Comment (chris-durbin) [inline ingest-app/docs/api.md]: Can you expand on "update value cannot contain arrays"? How would that be accomplished then - do they call update-value multiple times to add each value they want? Also capitalize update.

Comment (sxu123) [reply ingest-app/docs/api.md]: Yes. If you want to ADD_TO_EXISTING, with a list of platforms, you will have to add one at a time. If you want to CLEAR_ALL_AND_REPLACE, you can only replace with one platform.   DONE.

Comment (sxu123) [reply ingest-app/docs/api.md]: DONE.

File: ingest-app/docs/api.md [hunk 2]
```diff
@@ -581,40 +579,43 @@ The following update types are supported:
   * Find and update - merge update value into any instance in the list that matches the find value.
   * Find and remove - remove any instance from the list that matches the find value.
 
-Update types that include a find will match on the fields supplied. For example, for a science keyword update with a find value of {"Category": "EARTH SCIENCES"}, any science keyword with a category of "EARTH SCIENCES" will be considered a match regardless of the values of the science keyword topic, term, etc.
+Bulk update post request takes the following parameters:
+  
+  * Concept-ids (required) - a list of concept ids to update
+  * Update type (required) - choose from the enumeration: ADD_TO_EXISTING, CLEAR_ALL_AND_REPLACE, FIND_AND_REPLACE, FIND_AND_REMOVE, FIND_AND_UPDATE
```
Comment (oubiwann) [inline ingest-app/docs/api.md]: For this sorts of things (e.g., constant names), I like to use "code" markdown (backticks), such as `ADD_TO_EXISTING`, `CLEAR_ALL_AND_REPLACE`, `FIND_AND_REPLACE`, `FIND_AND_REMOVE`, `FIND_AND_UPDATE`

File: ingest-app/docs/api.md [hunk 3]
```diff
@@ -581,40 +579,43 @@ The following update types are supported:
...
+
+Update types that include a FIND will match on the fields supplied in the find value. For example, for a science keyword update with a find value of {"Category": "EARTH SCIENCE"}, any science keyword with a category of "EARTH SCIENCE" will be considered a match regardless of the values of the science keyword topic, term, etc.  It's worth noting that find value can not contain nested fields. So for bulk update on PLATFORMS, for example, find value can only contain Type, ShortName and LongName, not the nested fields like Characteristics and Instruments. On the other hand, update value can contain all the valid fields including the nested fields. So, nested fields can be updated, they just can't be used to find the matches. Please also note that update value can not be an array - say you want to use ADD_TO_EXISTING to add a list of platforms, you can only add one platform at a time. If you want to use CLEAR_ALL_AND_REPLACE to replace the existing platforms with a list of new platforms, it can NOT be achieved currently. You can only replace the existing platforms with one new platform.
```
Comment (oubiwann) [inline ingest-app/docs/api.md]: Could you add backticks around the data structure too? I.e., `{"Category": "EARTH SCIENCE"}`

**Pull request merged**